cmd/process-agent/collector.go

```diff
@@ -105,9 +105,43 @@ func (l *Collector) runCheck(c checks.Check, results *api.WeightedQueue) {
 	start := time.Now()
 	// update the last collected timestamp for info
 	updateLastCollectTime(start)
-	messages, err := c.Run(l.cfg, atomic.AddInt32(&l.groupID, 1))
-	if err != nil {
-		log.Errorf("Unable to run check '%s': %s", c.Name(), err)
+
+	if withRealTime, ok := c.(checks.CheckWithRealTime); ok {
+		run, err := withRealTime.RunWithOptions(l.cfg, l.nextGroupID, options)
+		if err != nil {
+			log.Errorf("Unable to run check '%s': %s", c.Name(), err)
+			return
+		}
+		l.messagesToResults(start, withRealTime.Name(), run.Standard, results)
+		l.messagesToResults(start, withRealTime.RealTimeName(), run.RealTime, results)
+	} else {
+		messages, err := c.Run(l.cfg, l.nextGroupID())
+		if err != nil {
+			log.Errorf("Unable to run check '%s': %s", c.Name(), err)
+			return
+		}
+		l.messagesToResults(start, c.Name(), messages, results)
+	}
+
+	if !c.RealTime() || options.RunStandard {
+		d := time.Since(start)
+		switch {
+		case runCounter < 5:
+			log.Infof("Finished %s check #%d in %s", c.Name(), runCounter, d)
```

Comment:
should we only increment `runCounter` when the normal check is run?

Reply:
We do check for it [here](https://github.com/DataDog/datadog-agent/pull/8846/files/08c387d53948737801e468cd267263f3b15968e2#diff-993d2491e7efac8e13349ef7859c375a68e328aa5415289b699a6da0cc2637c3R126)

Reply:
If `runCheck` is called for both standard and RT checks, won't `runCounter` be increased by RT checks? 
https://github.com/DataDog/datadog-agent/blob/08c387d53948737801e468cd267263f3b15968e2/cmd/process-agent/collector.go#L99-L103

Reply:
yes, but RT check in that case is an illusion because we are just copying stats from the standard check